Update atomic eligibility copy for unlaunched and private sites

p-jackson opened the pull request:
#### Changes proposed in this Pull Request

Part of the #38141, this PR only includes wording changes for telling the user their site must be launched. Just to keep the wording changes easier to review. The bigger style changes are in #38277

* Change message shown for _unlaunched_ site with `SITE_PRIVATE` hold.
* Change message shown for _launched_ site with `SITE_PRIVATE` hold.

*Message for unlaunched site*
<img width="653" alt="Screenshot 2019-12-11 at 10 14 40 AM" src="https://user-images.githubusercontent.com/1500769/70569949-d0af9600-1bff-11ea-8748-1472c597a6b4.png">

*Message for launched private site*
<img width="695" alt="Screenshot 2019-12-11 at 10 26 24 AM" src="https://user-images.githubusercontent.com/1500769/70570480-da85c900-1c00-11ea-9cbe-3d81fbaf899c.png">
The help link goes to https://en.support.wordpress.com/settings/privacy-settings/, which is currently what production does. The button will probably be removed as part of #38141 since proceeding will change the setting for the user.

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Choose a simple site to test that's unlaunched
  * Go to plugins, click upload plugin
  * Should see message that site must be launched
* Choose a simple simple site that's launched but private
  * Go to plugins, click upload plugin
  * Should see message that site must be made public

p-jackson committed:
Add business plan error to hold list

Also removed code for NOT_USING_CUSTOM_DOMAIN which is no longer
returned by the backend.

p-jackson committed:
Restyle hold list to use checkmarks

p-jackson committed:
AT blockers appear at the top of <EligibilityWarnings>

p-jackson committed:
Dim the fixable holds when a blocking hold is visible

p-jackson committed:
Remove business upgrade banner from <EligibilityWarnings>

We're showing that a business plan is required in the holds list for
consistency inside this component.

Removing banner allows some mocking to be removed from tests.

p-jackson committed:
Add aria-label prop to <NoticeAction>

This allows actions with only an icon to add a label for screen readers.

p-jackson committed:
Move AT warnings to top of UI, seperate from the holds area

p-jackson committed:
Change <EligibilityWarning> message if site is unlaunched

p-jackson committed:
Update message for launched private site

p-jackson commented:
Could you take a look at these copy changes @obi2020? The change for unlaunched sites is taken directly from paObgF-Mn-p2, but I also took a stab at wording for the case where a site is private and launched, which wasn't covered by the figma file.

p-jackson committed:
Remove unused styles

Styles unused now that warnings use <Notice>

p-jackson committed:
Refactor <EligibilityWarnings> to define the <Card> outline

p-jackson committed:
Increase space between heading and hold list

p-jackson committed:
Restyle eligibility warnings confirmation button

ramonjd reviewed (approved):
Does what it says on the tin.
Both messages appear. 
Do we need to string freeze this for translations, or do we have time?

ramonjd commented:
Not sure about the desktop CircleCI fails. I kicked them off again. 🤞

p-jackson commented:
@ramonjd This is probably all going to be merged into an uber PR so maybe we do a string freeze there so we can track translations for the whole feature?

obi2020 commented:
I think it's a little confusing. We say "make it public" then say that it will make the site "hidden" - it seems to be a contradiction. In the support page you link to this is shown:
![privacy-privacy-settings](https://user-images.githubusercontent.com/3373750/70634955-13be3900-1c01-11ea-851c-1e4326b879d3.png)

Which are we telling them to do?

obi2020 commented on client/blocks/eligibility-warnings/hold-list.tsx:
```diff
@@ -28,11 +28,17 @@ function getHoldMessages( translate: LocalizeProps[ 'translate' ] ) {
+						"This creates the basic layer the fancier stuff is built on. Don't worry, you can share this with everyone only when you're ready."
+				  )
+				: translate(
+						"Change's your site's Privacy settings to \"Hidden\". Don't worry, you can share this with everyone only when you're ready."
```
The header and description are contradictory: Says "make it public" then description says that was change settings to "hidden" - meanwhile these are show as two separate options in the support page that is linked. 

Need to clarify the second requirement point so user knows precisely which to choose.

p-jackson replied:
You're right that is confusing 😕 This will be a lot easier one search engine indexing is a checkbox under Public paObgF-MY-p2

p-jackson replied:
I've updated it and focused more on the visibility aspect instead, and copied some of the wording from the settings view.
<img width="730" alt="Screenshot 2019-12-12 at 4 18 11 PM" src="https://user-images.githubusercontent.com/1500769/70679990-a511d600-1cfb-11ea-98b8-a6acdd0a3e51.png">

obi2020 replied:
Thank you! That's definitely better, but still little unclear on whether they will need to take an action or this will happen automatically when they upgrade & install something. If they do need to take action, will they be prompted later?

p-jackson replied:
Making the site visible happens automatically when clicking "Upgrade and continue". Maybe it's confusing the heading says "you'll need a couple things" when really _you_ don't need to do anything (except confirm).

adamziel replied:
How about changing `Make site visible` to `A public site`?

adamziel replied:
also cc @ianstewart

michaeldcain replied:
One issue I see is that we say that the user will need things. Then the items are actions. We should make both items things.

> ### To install plugins, you'll need a couple of things:
> 
> **A Business plan**
> …
> 
> **A Public site**
> …

adamziel replied:
The root PR now says "You'll need to:" so the phrasing here would work

obi2020 reviewed (changes requested):
Clarify copy for launched sites

p-jackson committed:
Update copy for launched private site to be "visibility" focused

p-jackson committed:
Fix test that wasn't testing anything

The default window.location in Jest is example.com, so my test wasn't
doing anything. Doesn't look like jsdom actually follows links so the
best way to test a link will do the right thing is just to test its
href.

p-jackson committed:
typos

adamziel commented:
I nitpicked the copy a little bit. Since that's what this PR is all about I'd say we should resolve that conversation before merging.

adamziel committed:
Implement Ian's updates to the design

adamziel committed:
Update copy

adamziel committed:
Remove a comma from warnings title

adamziel commented:
Let's merge it as it is once the layout change lands here: https://github.com/Automattic/wp-calypso/pull/38277 - it's infinitely better than what we currently have in production. Let's make updating the copy even further v2

p-jackson committed:
Fixup tests after changes to <WarningList>

p-jackson committed:
Copy for when component appears in /themes and /hosting-config contexts

adamziel committed:
Remove icons for warnings list

adamziel committed:
Update CSS classes to set entire warnings/holds list into indented/unindented mode (vs specific messages)

adamziel commented:
Rebasing this to master will fix most conflicts

adamziel commented:
This is likely not relevant anymore since we'll release private sites soon. We'll revisit if necessary.
